pkg/virt-handler/heartbeat/heartbeat.go

```diff
@@ -103,6 +103,21 @@ func (h *HeartBeat) labelNodeUnschedulable() (done chan struct{}) {
 	return done
 }
 
+func (h *HeartBeat) isMasterNodeWorker() bool {
+	node, err := h.clientset.Nodes().Get(context.Background(), h.host, metav1.GetOptions{})
+	if err != nil {
+		log.DefaultLogger().Reason(err).Errorf("Can't get node %s", h.host)
+	}
+
+	//check if node is master node, return true if no master node label
+	_, ok := node.Labels["node-role.kubernetes.io/master"]
+	if !ok {
+		return true
+	}
+	_, ok = node.Labels["node-role.kubernetes.io/worker"]
+	return ok
+}
```

Comment:
- Can't we just check if the node is a worker? If we need to consider the "master" label, please also include `node-role.kubernetes.io/control-plane`.
- We don't use the term "master" anymore in our code/comments. Please use "control-plane" instead. (the label name is obviously a valid exception)
- What about node taints? If a node is a worker but has a NoSchedule taint, shouldn't we set `kubevirt.io/schedulable` to false?

Reply:
Will update to use control-plane. I also wasn't sure about just checking for the worker role but I suppose I don't think there are any instances where a node would not have a worker label and be schedulable, so I can modify it to just check for the worker role.

At this point I believe a worker node with a NoSchedule taint also remains with schedulable=true so I can set schedulable to false for that case as well.

Reply:
I've added another function that checks for the NoSchedule taint before setting schedulable to true/false.